src/node.cc

```diff
@@ -237,23 +237,15 @@ class NodeTraceStateObserver :
     trace_process->SetString("napi", node_napi_version);
 
 #if HAVE_OPENSSL
-    // Stupid code to slice out the version string.
-    {  // NOLINT(whitespace/braces)
-      size_t i, j, k;
-      int c;
-      for (i = j = 0, k = sizeof(OPENSSL_VERSION_TEXT) - 1; i < k; ++i) {
-        c = OPENSSL_VERSION_TEXT[i];
-        if ('0' <= c && c <= '9') {
-          for (j = i + 1; j < k; ++j) {
-            c = OPENSSL_VERSION_TEXT[j];
-            if (c == ' ')
-              break;
-          }
-          break;
-        }
-      }
-      trace_process->SetString("openssl",
-                              std::string(&OPENSSL_VERSION_TEXT[i], j - i));
+    {
+      // sample openssl version string format
+      // for reference: "OpenSSL 1.1.0i  14 Aug 2018"
+      std::string ssl(OPENSSL_VERSION_TEXT);
```

Comment:
Could you wrap the first 5 lines into a `constexpr string()(const char*)` and reuse in the second place?

Reply:
@refack - sure, given `OPENSSL_VERSION_TEXT` is accessible everywhere (in the caller and the callee) do you mind if I don't pass it? are there any guidance on that? thanks.

Reply:
Yes, of course, but I gave it a try, and we might need C++17 for this to work as expected.
But is could definatly be `inline`